### PR TITLE
Update `getTotalSupply` function in LIP 0051, add pseudocode to endpoint, clarify store logic

### PR DESCRIPTION
**Pull request by mjerkov**
Resolves Issue #448.

Additionally, following two changes are proposed:
1. In the [Store notation](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0051.md#store-notation) subsection, the statement "Whenever a function calls property of a non-existing substore entry, an error is thrown." is changed to "Whenever a queried substore entry does not exist, an error is thrown.", in order to conform to the description and original intention of [PR 259](https://github.com/LiskHQ/lips/pull/259), where it was introduced.
2. The pseudocode for `getSupportedTokens` endpoint has been added.

### CODE REVIEW

**Review by sergeyshemyakov — APPROVED** [review]

File: proposals/lip-0051.md [hunk 1]
```diff
@@ -2635,6 +2635,35 @@ This section specifies the non-trivial or recommended endpoints of the Token mod
...
 This function returns token IDs for both native and non-native tokens supported by the chain.
 
+```python
+def getSupportedTokens() -> dict[list]:
+    # Native tokens
+    nativeTokenIDs = [tokenID for tokenID in supplyStore.keys()]
+
+    # Non-native tokens	
+    if supportedTokens(ALL_SUPPORTED_TOKENS_KEY) exists:
+        return {
```
Comment (sergeyshemyakov) [inline proposals/lip-0051.md:2645]: In order to be more aligned with the SDK I would return `{['*']}` here.

File: proposals/lip-0051.md [hunk 2]
```diff
@@ -2635,6 +2635,35 @@ This section specifies the non-trivial or recommended endpoints of the Token mod
...
+            "nativeTokenIDs": nativeTokenIDs,
+            "supportedChainIDs": "All",
+            "supportedTokenIDs": "All"
+        }
+
+    supportedChainIDs = []
+    supportedTokenIDs = [getMainchainID()]
```
Comment (sergeyshemyakov) [inline proposals/lip-0051.md:2652]: In order to be more aligned with the SDK I would write `supportedTokenIDs = [getMainchainID() + b'\x00'*4]` here. The implementation pads the mainchain ID with zeros to the length of a token ID to get the token ID of LSK in the current network.

File: proposals/lip-0051.md [hunk 3]
```diff
@@ -2635,6 +2635,35 @@ This section specifies the non-trivial or recommended endpoints of the Token mod
...
+    
+    for chainID in supportedTokens.keys():
+        if supportedTokens(chainID).supportedTokenIDs == []:
+            supportedChainIDs.append(chainID)
```
Comment (sergeyshemyakov) [inline proposals/lip-0051.md:2656]: In order to be more aligned with the SDK I would write here `supportedTokenIDs.append(chainID+"********")`.

File: proposals/lip-0051.md [hunk 4]
```diff
@@ -2635,6 +2635,35 @@ This section specifies the non-trivial or recommended endpoints of the Token mod
...
+        else:
+            supportedTokenIDs.extend(supportedTokens(chainID).supportedTokenIDs)
+
```
Comment (sergeyshemyakov) [inline proposals/lip-0051.md:2659]: In order to be more aligned with the SDK I would add a line `supportedTokenIDs.extend(nativeTokenIDs)`.
An alternative would be to discard the variable `nativeTokenIDs` altogether and write into `supportedTokenIDs` from the very beginning.

File: proposals/lip-0051.md [hunk 5]
```diff
@@ -2635,6 +2635,35 @@ This section specifies the non-trivial or recommended endpoints of the Token mod
...
+    return {
```
Comment (sergeyshemyakov) [inline proposals/lip-0051.md:2660]: In order to be more aligned with the SDK I would return just `supportedTokenIDs`.

**Review by sergeyshemyakov — COMMENTED** [review]
I think it would be good to align the `getSupportedTokens` endpoint with the SDK, otherwise I don't see why we have it in the LIP in the first place.

**Review by sergeyshemyakov — APPROVED** [review]
Looks good!

**Review by janhack — APPROVED** [review]
Looks all good.

File: proposals/lip-0051.md [hunk 6]
```diff
@@ -1598,10 +1598,12 @@ def getLockedAmount(address: Address, module: Module, tokenID: TokenID) -> uint6
...
 ```python
-def getTotalSupply(tokenID: TokenID) -> uint64:
-     if totalSupply(tokenID) does not exist:
-         raise Exception("Total supply entry does not exist.")
-     return totalSupply(tokenID)
+def getTotalSupply() -> dict[TokenID, uint64]:
+    totalSupply: dict[TokenID, uint64] = {}
+    for (storeKey, storeValue) in supplyStore.items():
+        tokenID = storeKey
+        totalSupply[tokenID] = storeValue.totalSupply
```
[the remark this answers is not in the document]
Comment (mjerkov) [reply proposals/lip-0051.md:1605]: Thanks for the suggestion, will do in the new PR.

File: proposals/lip-0051.md [hunk 7]
```diff
@@ -1598,10 +1598,12 @@ def getLockedAmount(address: Address, module: Module, tokenID: TokenID) -> uint6
 #### getTotalSupply
 
 ```python
-def getTotalSupply(tokenID: TokenID) -> uint64:
-     if totalSupply(tokenID) does not exist:
-         raise Exception("Total supply entry does not exist.")
-     return totalSupply(tokenID)
+def getTotalSupply() -> dict[TokenID, uint64]:
+    totalSupply: dict[TokenID, uint64] = {}
+    for (storeKey, storeValue) in supplyStore.items():
```
[the remark this answers is not in the document]
Comment (mjerkov) [reply proposals/lip-0051.md:1603]: I opted out for using the version `for tokenID in supplyStore.keys()`, following the info from this [site](https://realpython.com/iterate-through-dictionary-python/#iterating-through-keys).

File: proposals/lip-0051.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (mjerkov) [reply proposals/lip-0051.md:1605]: Please check [this PR](https://github.com/LiskHQ/lips/pull/458).

**Pull request merged**